### PR TITLE
Add analysis/cross_tabs

### PR DESCRIPTION
**Pull request by as2388**
Sample output: https://drive.google.com/file/d/1wXY7rFIl6joWGifs3dDx649xd5-4gSkt/view?usp=sharing

### CODE REVIEW

File: core_data_modules/analysis/cross_tabs.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    # Initialise the cross_tab_counts dict with all the possible combinations in the 2 given code schemes, with counts
+    # initially set to 0.
+    for code_1 in _normal_codes(analysis_configuration_1.code_scheme.codes):
+        for code_2 in _normal_codes(analysis_configuration_2.code_scheme.codes):
+            # Assign the count dict for this pair of codes
+            # e.g. { "gender": "man", "age_category": "10_to_14", "Number of Individuals": 0 }
+            cross_tab_counts[(code_1.string_value, code_2.string_value)] = {
+                analysis_configuration_1.dataset_name: code_1.string_value,
+                analysis_configuration_2.dataset_name: code_2.string_value,
+                "Number of Individuals": 0
```
Comment (IsaackMwenda) [inline core_data_modules/analysis/cross_tabs.py:51]: consider moving this to constant instead of duplicating the literal.

**Review by IsaackMwenda — APPROVED** [review]
LGTM

File: core_data_modules/analysis/cross_tabs.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        assert len(config_2_normal_codes) <= 1, config_2_normal_codes
+
+        if len(config_1_normal_codes) == 0 or len(config_2_normal_codes) == 0:
+            continue
+
+        cross_tab_key = (config_1_normal_codes[0].string_value, config_2_normal_codes[0].string_value)
+        cross_tab_counts[cross_tab_key]["Number of Individuals"] += 1
+
+    # Compute percentages
+    for code_1 in _normal_codes(analysis_configuration_1.code_scheme.codes):
```
Comment (DanielMwendwa) [inline core_data_modules/analysis/cross_tabs.py:71]:
```suggestion
    for analysis_config_1_code in _normal_codes(analysis_configuration_1.code_scheme.codes):
```

File: core_data_modules/analysis/cross_tabs.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    for ind in individuals:
+        config_1_normal_codes = _normal_codes(get_codes_from_td(ind, analysis_configuration_1))
+        config_2_normal_codes = _normal_codes(get_codes_from_td(ind, analysis_configuration_2))
+
+        # Cross-tab analysis only supports individuals which have at most 1 normal code for now.
+        assert len(config_1_normal_codes) <= 1, config_1_normal_codes
+        assert len(config_2_normal_codes) <= 1, config_2_normal_codes
+
+        if len(config_1_normal_codes) == 0 or len(config_2_normal_codes) == 0:
+            continue
```
Comment (DanielMwendwa) [inline core_data_modules/analysis/cross_tabs.py:65]: Consider moving L60-62 after the if statement and then assert `len(config_1_normal_codes) == 1`

File: core_data_modules/analysis/cross_tabs.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            cross_tab_counts[(code_1.string_value, code_2.string_value)] = {
+                analysis_configuration_1.dataset_name: code_1.string_value,
+                analysis_configuration_2.dataset_name: code_2.string_value,
+                "Number of Individuals": 0
+            }
+
+    # Count up all the individuals in each category.
+    individuals = analysis_utils.filter_opt_ins(individuals, consent_withdrawn_field)
+    for ind in individuals:
+        config_1_normal_codes = _normal_codes(get_codes_from_td(ind, analysis_configuration_1))
```
Comment (DanielMwendwa) [inline core_data_modules/analysis/cross_tabs.py:57]:
```suggestion
        config_1_normal_codes = _normal_codes(analysis_utils.get_codes_from_td(ind, analysis_configuration_1))
```

Comment (DanielMwendwa) [reply core_data_modules/analysis/cross_tabs.py:57]: Update the imports if you agree.

**Review by DanielMwendwa — APPROVED** [review]

File: core_data_modules/analysis/cross_tabs.py (hunk 1 above)
Comment (as2388) [reply core_data_modules/analysis/cross_tabs.py:51]: updated

File: core_data_modules/analysis/cross_tabs.py (hunk 3 above)
Comment (as2388) [reply core_data_modules/analysis/cross_tabs.py:65]: updated

File: core_data_modules/analysis/cross_tabs.py (hunk 2 above)
Comment (as2388) [reply core_data_modules/analysis/cross_tabs.py:71]: Renamed to config_1_code, for consistency with the naming used in the loop above

**Pull request merged**